lib/grape/jsonapi/services/content_negotiator.rb

```diff
@@ -0,0 +1,50 @@
+module Grape
+  module Jsonapi
+    module Services
+      class ContentNegotiator
+        VALID_MEDIA_TYPE = 'application/vnd+json'.freeze
+
+        ERRORS = {
+          not_acceptable: {
+            status: 406,
+            message: 'Not Acceptable'
+          },
+          unsupported_media: {
+            status: 415,
+            message: 'Unsupported Media Type'
+          }
+        }.freeze
+
+        def initialize(api_version, headers)
+          @api_version    = api_version
+          @accept_version = headers['Accept-Version']
+          @content_type   = headers['Content-Type']
+          @accept_header  = headers['Accept']
+        end
+
+        def self.run(*args)
+          new(*args).run
+        end
+
+        def run
+          return ERRORS[:unsupported_media] unless valid?(content_type)
+          return ERRORS[:not_acceptable] unless valid?(accept_header)
+
+          valid_accept_version_header?
```

Comment:
lets remove all concept of versioning and accept-version header from this service object and only focus on the content-type header